src/group_theory/free_group.lean

```diff
@@ -369,10 +369,22 @@ instance : has_mul (free_group α) :=
     (λ L₁ L₂ H, quot.induction_on y $ λ L₃, quot.sound $ red.step.append_right H)⟩
 @[simp] lemma mul_mk : mk L₁ * mk L₂ = mk (L₁ ++ L₂) := rfl
 
+/-- Transform a word representing a free group element into a word representing its inverse. --/
+def inv_rev (w : list (α × bool)) : list (α × bool) :=
+(list.map (λ (g : α × bool), (g.1, bnot g.2)) w).reverse 
+
+@[simp] lemma inv_rev_eq (w : list (α × bool)) : inv_rev w =  
+(list.map (λ (g : α × bool), (g.1, bnot g.2)) w).reverse := rfl 
+
+lemma inv_rev_length : (inv_rev L₁).length = L₁.length :=
+begin
+  induction L₁, { finish, },
+  unfold inv_rev, { finish, },
```

Comment:
```suggestion
  induction L₁, 
  { refl },
  { simp }
```
We're trying to keep `finish` out of mathlib, since it will be hard to port exactly to Lean 4. In any case, it's overkill here!